### DISCUSSION TITLE
OEM print development

### DISCUSSION THREAD

**Discussion by abadiag** [microsoft/WindowsAppSDK#2421] [Dev team discussions]
Hello, I am a developer in a printers OEM, we are trying to adapt our UI to the old metro app (more settings). We run into the problem that the window that opens through the metadata link is from the UWP context, so winUI throws the exception :  _'WinUI: Error creating an UWP Window. Creating an UWP window is not allowed.'.

Are you planning to enable to create a UWP compat window or something like this?
How can we solve this issue?

![Diagram](https://user-images.githubusercontent.com/38919412/164214696-bfd8fe64-1bd4-4892-b413-24b6aa588e5b.png)

_With experimental package it was possible because framework include universal platform_

**Comment by angelazhangmsft** [comment]
@abadiag - Printing is currently not supported in WinUI3 desktop apps. We are planning to migrate the functionality to WinUI3 in a future release. Are you migrating an app from UWP to WinUI3 and trying to print out WinUI XAML controls?

**Reply by abadiag** [reply]
I'm a developer on the Drivers team,
we're trying to save the current driver backend by using WinUI3 to run the win32 backend and just implement new controls. I see that the app starts, but in the LaunchActivatedEventArgs I get "Launch" not "print" like the old version, in addition to giving me the error that I mentioned at the beginning.

Do you know when the printing part will be implemented?

this is the microsoft official link that we follow on the past development:
[how-to-customize-print-settings](https://docs.microsoft.com/en-us/windows-hardware/drivers/devapps/how-to-customize-print-settings)

**Reply by castorix** [reply]
> @abadiag - Printing is currently not supported in WinUI3 desktop apps. 

Printing can be done in WinUI3 with [PrintDocument ](https://docs.microsoft.com/en-us/dotnet/api/system.drawing.printing.printdocument?view=dotnet-plat-ext-6.0)

**Reply by abadiag** [reply]
We don't trying to print from WinUI3 app, we are making the UI of the driver
